meta/saimetadatautils.h

```diff
@@ -229,6 +229,99 @@ extern bool sai_metadata_is_validonly_met(
  */
 extern sai_api_version_t sai_metadata_query_api_version(void);
 
+**
+ * @brief Allocation info
+ *
+ * hint for the allocation
+ */
+typedef struct _sai_alloc_info_t
+{
+    uint32_t list_size;
+
+    /**
+     * @brief Reference attribute for size information
+     */
+    const sai_attribute_t *reference;
+} sai_alloc_info_t;
+
+/**
+ * @brief Allocate sai_attribute_t value
+ *
+ * By passing info == NULL, it will use default list size for the list value
+ * allocation
+ *
+ * @param[in] metadata Attribute metadata
+ * @param[in] attr Attribute to allocate
+ * @param[in] info Allocation information
+ *
+ * @return #SAI_STATUS_SUCCESS on success,
+ * #SAI_STATUS_INVALID_PARAMETER/#SAI_STATUS_NO_MEMORY on failure
+ */
+extern sai_status_t sai_metadata_alloc_attr_value(
+        _In_ const sai_attr_metadata_t *metadata,
+        _In_ const sai_attribute_t *attr,
+        _In_ const sai_alloc_info_t *info);
+
+/**
+ * @brief Free sai_attribute_t value
+ *
+ * @param[in] metadata Attribute metadata
+ * @param[in] attr Attribute to free
+ * @param[in] info Allocation information
+ *
+ * @return #SAI_STATUS_SUCCESS on success,
+ * #SAI_STATUS_INVALID_PARAMETER on failure
+ */
+extern sai_status_t sai_metadata_free_attr_value(
+        _In_ const sai_attr_metadata_t *metadata,
+        _In_ const sai_attribute_t *attr,
+        _In_ const sai_alloc_info_t *info);
+
+/**
+ * @brief Clear sai_attribute_t value
+ *
+ * @param[in] metadata Attribute metadata
+ * @param[in] attr Attribute to clear
+ *
+ * @return #SAI_STATUS_SUCCESS on success,
+ * #SAI_STATUS_INVALID_PARAMETER on failure
+ */
+extern sai_status_t sai_metadata_clear_attr_value(
+        _In_ const sai_attr_metadata_t *metadata,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Deep copy sai_attribute_t value
+ *
+ * @param[in] metadata Attribute metadata
+ * @param[in] src Original attribute for the copy
+ * @param[inout] dest Destination for the copy
+ *
+ * @return #SAI_STATUS_SUCCESS on success,
+ * #SAI_STATUS_INVALID_PARAMETER on failure
+ */
+extern sai_status_t sai_metadata_deepcopy_attr_value(
+        _In_ const sai_attr_metadata_t *metadata,
+        _In_ const sai_attribute_t *src,
+        _Inout_ sai_attribute_t *dest);
+
+/**
+ * @brief Deep equal sai_attribute_t value
+ *
+ * @param[in] metadata Attribute metadata
+ * @param[in] lhs Left
+ * @param[in] rhs Right
+ * @param[out] result Result
+ *
+ * @return #SAI_STATUS_SUCCESS on success,
+ * #SAI_STATUS_INVALID_PARAMETER on failure
+ */
+extern sai_status_t sai_metadata_deepequal_attr_value(
+        _In_ const sai_attr_metadata_t *metadata,
+        _In_ const sai_attribute_t *lhs,
+        _In_ const sai_attribute_t *rhs,
+        _Out_ bool *result);
+		
```

Comment:
those functions are not in utils and seems like should not be at all, not sure what they do, maybe only for internal use